### PR TITLE
fix(network-manager): support teaming under NM+systemd

### PR DESCRIPTION
**Pull request by dustymabe**
Previously when NM was run without dbus then teaming would come
up appropriately [1], but now that dbus exists we also need to
include some supporting infrastructure to allow for it to work
again.

[1] https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/commit/d689380cfc5734a29b1302d68027190e1a606265

### CODE REVIEW

**Comment by dustymabe** [comment]
I have tested this locally.

File: modules.d/35network-manager/module-setup.sh
Comment (dustymabe) [inline modules.d/35network-manager/module-setup.sh:45]: @jpirko - the fedora rpm currently defines the dbus conf in `/etc/dbus-1/system.d/teamd.conf`. Should it be modified to place it in `/usr/share/dbus-1/system.d/teamd.conf` instead?

Here I'm picking it up from either place in case it changes.

**Comment by dustymabe** [comment]
cc @bengal

File: modules.d/35network-manager/module-setup.sh
Comment (bengal) [inline modules.d/35network-manager/module-setup.sh:43]: Currently, NetworkManager spawns teamd directly and then waits that teamd appears on D-Bus to communicate with it.
This line above is not needed because teamd is not started as a systemd service.

The rest LGTM, I also tried the patch and it works well. Thanks!

Comment (dustymabe) [reply modules.d/35network-manager/module-setup.sh]: fixed in latest push. Thanks @bengal.

**Comment by dustymabe** [comment]
@haraldh - when we merge this do you think we could get it backported to F34?

File: modules.d/35network-manager/module-setup.sh
Comment (bengal) [reply modules.d/35network-manager/module-setup.sh]: LGTM now.

**Comment by dustymabe** [comment]
> LGTM now.

Thanks @bengal - any other reviewers we need to wait on or can this be merged now?

**Review by johannbg — APPROVED** [review]
LGTM

**Comment by dustymabe** [comment]
Any ideas how to fix the failing `packit` test?

**Comment by johannbg** [comment]
@dustymabe packit tests passing are not a requirement so you can just ignore it.  Anyways to fix a failing packit test requires us to open up an issue with the packit team, which then will look into it along with infra with high probability that it's not our PR's causing failure, not packit itself but downstream maintainer or infra making some mistake or upgrade ( this has happened before ).

**Review by haraldh — APPROVED** [review]

**Pull request merged**
